Helper/Cart.php

```diff
@@ -305,31 +316,187 @@ public function getOrderByIncrementId($incrementId)
     /**
      * Save quote via repository
      *
-     * @param \Magento\Quote\Api\Data\CartInterface $quote
+     * @param CartInterface $quote
      */
     public function saveQuote($quote)
     {
         $this->quoteRepository->save($quote);
     }
 
+    /**
+     * Delete quote via repository
+     *
+     * @param CartInterface $quote
+     */
+    public function deleteQuote($quote)
+    {
+        $this->quoteRepository->delete($quote);
+    }
+
     /**
      * Save quote via resource model
      *
-     * @param \Magento\Quote\Api\Data\CartInterface $quote
+     * @param CartInterface $quote
      * @throws \Magento\Framework\Exception\AlreadyExistsException
      */
     public function quoteResourceSave($quote)
     {
         $this->quoteResource->save($quote);
     }
 
+    /**
+     * Get Bolt Order Caching flag configuration
+     *
+     * @param null|int    $storeId             The ID of the Magento store
+     * @return bool
+     */
+    protected function isBoltOrderCachingEnabled($storeId)
+    {
+        return $this->configHelper->isBoltOrderCachingEnabled($storeId);
+    }
+
+    /**
+     * Load data from Magento cache
+     *
+     * @param string $identifier
+     * @param bool $unserialize
+     * @return bool|mixed|string
+     */
+    protected function loadFromCache($identifier, $unserialize = true)
+    {
+        $cached = $this->cache->load($identifier);
+        if (!$cached) return false;
+        return $unserialize ? unserialize($cached) : $cached;
+    }
+
+    /**
+     * Save data to Magento cache
+     *
+     * @param mixed $data
+     * @param string $identifier
+     * @param int $lifeTime
+     * @param bool $serialize
+     * @param array $tags
+     */
+    protected function saveToCache($data, $identifier, $lifeTime = null, $serialize = true, $tags = [])
+    {
+        $data = $serialize ? serialize($data) : $data;
+        $this->cache->save($data, $identifier, $tags, $lifeTime);
+    }
+
+    /**
+     * Last created immutable quote setter
+     *
+     * @param $quote
+     */
+    protected function setLastImmutableQuote($quote)
+    {
+        $this->lastImmutableQuote = $quote;
+    }
+
+    /**
+     * Last created immutable quote getter
+     *
+     * @return CartInterface
+     */
+    protected function getLastImmutableQuote()
+    {
+        return $this->lastImmutableQuote;
+    }
+
+    /**
+     * Cache the session id for the quote
+     *
+     * @param array $cart
+     */
+    protected function saveCartSession($cart)
+    {
+        $this->sessionHelper->saveSession($cart['order_reference'], $this->checkoutSession);
```

Comment:
Can we refactor it? Just transfer a `$cart['order_reference']` to the `saveCartSession()` method instead of whole object `$cart`.

Reply:
done